modules/client/src/client-common/interfaces/plugin.ts

```diff
@@ -243,3 +242,42 @@ export type SubgraphVotingSettings = {
   supportThreshold: string;
   votingMode: VotingMode;
 };
+
+export enum PrepareInstallationStep {
+  PREPARING = "preparing",
+  DONE = "done",
+}
+
+export type PrepareInstallationStepValue =
+  | { key: PrepareInstallationStep.PREPARING; txHash: string }
+  | {
+    key: PrepareInstallationStep.DONE;
+  } & ApplyInstallationParams;
+
+export type ApplyInstallationParamsBase = {
+  permissions: MultitargetPermission[];
+  versionTag: VersionTag;
+  pluginRepo: string;
+  pluginAddress: string;
+}
+
+
+export type ApplyInstallationParams = ApplyInstallationParamsBase & {
+  helpers: string[]
+};
+export type DecodedApplyInstallationParams = ApplyInstallationParamsBase & {
+  helpersHash: string
+};
+
+export type VersionTag = {
+  build: number;
+  release: number;
+};
+
+export type MultitargetPermission = {
```

Comment:
I suggest we follow the naming and formatting from the core as closely as possible
```suggestion
export type MultiTargetPermission = {
```